stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -1022,36 +1022,19 @@ void SetUpInterfacesInterfaceStateLoopbackMode(uint64 node_id, uint32 port_id,
 
 ////////////////////////////////////////////////////////////////////////////////
 // /interfaces/interface[name=<name>]/state/hardware-port
-void SetUpInterfacesInterfaceStateHardwarePort(uint64 node_id, uint32 port_id,
-                                               TreeNode* node,
+void SetUpInterfacesInterfaceStateHardwarePort(TreeNode* node,
                                                YangParseTree* tree) {
-  // Regular method using a template cannot be used to get the OnPoll functor as
-  // std::string fields are treated differently by the PROTO-to-C++ generator:
-  // the getter returns "const std::string&" instead of "string" which leads to
-  // the template compilation error.
-  auto poll_functor = [node_id, port_id, tree](const GnmiEvent& event,
-                                      const ::gnmi::Path& path,
-                                      GnmiSubscribeStream* stream) {
-    // Create a data retrieval request.
-    DataRequest req;
-    auto* request = req.add_requests()->mutable_hardware_port();
-    request->set_node_id(node_id);
-    request->set_port_id(port_id);
-    // In-place definition of method retrieving data from generic response
-    // and saving into 'resp' local variable.
-    std::string resp{};
-    DataResponseWriter writer([&resp](const DataResponse& in) {
-      if (!in.has_hardware_port()) return false;
-      resp = in.hardware_port().name();
-      return true;
-    });
-    // Query the switch. The returned status is ignored as there is no way to
-    // notify the controller that something went wrong. The error is logged when
-    // it is created.
-    tree->GetSwitchInterface()
-        ->RetrieveValue(node_id, req, &writer, /* details= */ nullptr)
-        .IgnoreError();
-    return SendResponse(GetResponse(path, resp), stream);
+  // This leaf is a reference to the /components/component[name=<name>]/name
+  // leaf. We look up the data from there.
+  std::string component_name = node->GetPath().elem(1).key().at("name");
+  // TODO(max): the path should end with /name, but we don't support that yet.
+  ::gnmi::Path ref_path = GetPath("components")("component", component_name)();
```

Comment:
can we move these two lines into the functor?

Reply:
Probably, I didn't want to capture the node pointer in the closure, because lifetime?

If we move it into the functor, the ref path is created on every call, but it never changes.

Reply:
Alternatively we pass the name as string to the entire function and capture that.

Reply:
I see, I forgot that we need to pass other things like the tree to the functor if we move these two lines into the functor.
I think it is fine to keep them outside the functor